.github/workflows/pokeone-api-pr.yaml

```diff
@@ -23,17 +23,17 @@ jobs:
     - name: Begin Sonar Scan
       run: |
         dotnet tool install --global dotnet-sonarscanner
-        dotnet sonarscanner begin /k:"poke-one-web-api" /o:"poke-one-web" /d:sonar.login="${{ secrets.SONAR_TOKEN_API }}" /d:sonar.host.url="https://sonarcloud.io"
+        dotnet sonarscanner begin /k:"poke-one-web-api" \
+                                  /o:"poke-one-web" \
+                                  /d:sonar.login="${{ secrets.SONAR_TOKEN_API }}" \
+                                  /d:sonar.host.url="https://sonarcloud.io" \
+                                  /d:sonar.cs.vstest.reportsPaths=TestResults/*.trx \
+                                  /d:sonar.cs.opencover.reportsPaths=TestResults/*/coverage.opencover.xml
 
-    - name: Install dependencies
-      run: dotnet restore
-    - name: Build
-      run: dotnet build PokeOneWeb --configuration Release --no-restore
+    - name: Build & Install Dependencies
+      run: dotnet build PokeOneWeb --configuration Release
     - name: Test
-      run: dotnet test --no-restore --verbosity normal
-
-    #- name: Unit-Test Api
-    #  run: dotnet test PokeOneWeb --no-build --no-restore --verbosity normal -p:CollectCoverage=true -p:CoverletOutputFormat=opencover
+      run: dotnet test --collect "XPlat Code Coverage" --results-directory TestResults/ --logger "trx;LogFileName=unittests.trx" -- DataCollectionRunSettings.DataCollectors.DataCollector.Configuration.Format=opencover
```

Comment:
For test coverage results I would suggest to **exclude** the following directories:
- all test projects (only measure coverage on production code, test code is anway always 100% covered)
- all generated code. This includes the DB migrations and snapshots, located in PokeOneWeb.Data/Migrations.

Reply:
Good point, as far as I saw it the Test projects are excluded by default. And the Migration files are excluded by the project settings on SonarCloud. Files that are marked auto-generated are overall excluded, but I think some Migration files are not marked liked this automatically (as far as I saw).
I rather configure such things on SonarCloud directly so we do not bloat our pipelines. But this is just my preference. We can also change this in the future.